packages/grafana-ui/src/components/GraphNG/nullInsertThreshold.ts

```diff
@@ -87,11 +87,13 @@ function nullInsertThreshold(refValues: number[], frameValues: any[][], threshol
     prevValue = curValue;
   }
 
+  refValuesNew.push(getInsertValue(prevValue, prevValue, threshold));
+
   const filledLen = refValuesNew.length;
 
-  if (filledLen === len) {
-    return frameValues;
-  }
+  // if (filledLen === len) {
+  //   return frameValues;
+  // }
```

Comment:
IIUC, this means that every query from prometheus will now hit this condition :thinking:   that seems less than great, but maybe it does not matter much.  

Alternatively we could pass in the end time also and check if the last value is close enough

Reply:
> Alternatively we could pass in the end time also and check if the last value is close enough

yeah, i think this is gonna be the way to go. we cannot assume that there is an absent interval at the end unless we explicitly got no data back when querying the timerange which includes where it would have been expected.